### PR TITLE
[iOS][non-icu] HybridGlobalization fix localized DisplayName and LanguageName

### LINKED ISSUE #95427: [iOS] CultureInfo.DisplayName only shows abbreviation
### Description

This was originally reported here: https://github.com/dotnet/runtime/issues/70028. That issue has been closed, but the problem persists. `CultureInfo.DisplayName` does not return the localized name. I have enabled `HybridGlobalization`

### Reproduction Steps

`new CultureInfo("it-IT").DisplayName`

### Expected behavior

This should output "**Italienisch (Italien)**" when running on a device that has **German** set as primary device language.

### Actual behavior

It outputs "it (IT)".

### Regression?

_No response_

### Known Workarounds

I can use `CultureInfo.EnglishName` but this gives me the English name and not the localized name

### Configuration

.NET 8 on iOS 17 (simulator + physical device)
`<HybridGlobalization>true</HybridGlobalization>` in .csproj

### Other information

_No response_

### PR DESCRIPTION
**Pull request by mkhamoyan**
Fix localized DisplayName and LanguageName for hybrid mode and enable tests on Apple mobile platforms.

Fixes https://github.com/dotnet/runtime/issues/95427
cc @SamMonoRT

### CODE REVIEW

**Review by steveisok — APPROVED** [review]

File: src/native/libs/System.Globalization.Native/pal_locale.m [hunk 1]
```diff
@@ -781,24 +793,31 @@ int32_t GlobalizationNative_GetLocalesNative(UChar* value, int32_t length)
...
-        NSLocale *currentLocale = [NSLocale currentLocale];
-        NSString *localeName = @"";
-
-        if (!currentLocale)
-        {
-            return strdup([localeName UTF8String]);
-        }
-
-        if ([currentLocale.languageCode length] > 0 && [currentLocale.countryCode length] > 0)
+        if (NSLocale.preferredLanguages.count > 0)
```
Comment (mdh1418) [inline src/native/libs/System.Globalization.Native/pal_locale.m:796]: Can this ever be 0? From my OSX machine, it doesn't seem like I can make my Preferred Languages list empty.

Comment (steveisok) [reply src/native/libs/System.Globalization.Native/pal_locale.m:796]: I don't think so, but better to check.

Comment (mkhamoyan) [reply src/native/libs/System.Globalization.Native/pal_locale.m:796]: Shouldn't be empty, but I found these kind of issues https://github.com/OneSignal/OneSignal-iOS-SDK/issues/201 , decided to have the check.

**Review by mdh1418 — APPROVED** [review]
Looks good to me!

**Pull request merged**